modules/rostests/kmtests/ntos_io/IoDeviceInterface.c

```diff
@@ -1,125 +1,236 @@
-
-   for (SymbolicLinkListPtr = SymbolicLinkList;
-        SymbolicLinkListPtr[0] != 0 && SymbolicLinkListPtr[1] != 0;
-        SymbolicLinkListPtr += wcslen(SymbolicLinkListPtr) + 1)
-   {
-      UNICODE_STRING SymbolicLink;
-      UNICODE_STRING AliasSymbolicLink;
-
-      SymbolicLink.Buffer = SymbolicLinkListPtr;
-      SymbolicLink.Length = SymbolicLink.MaximumLength = wcslen(SymbolicLinkListPtr);
-      RtlInitUnicodeString(&AliasSymbolicLink, NULL);
-      IoGetDeviceInterfaceAlias(
-         &SymbolicLink,
-         &AliasGuid,
-         &AliasSymbolicLink);
-      if (AliasSymbolicLink.Buffer != NULL)
-      {
-         DPRINT("[PnP Test] Original: %S\n", SymbolicLinkListPtr);
-         DPRINT("[PnP Test] Alias: %S\n", AliasSymbolicLink.Buffer);
-      }
-   }
-#endif
-
-   ExFreePool(SymbolicLinkList);
+    &GUID_NULL,
+    &KSCATEGORY_BRIDGE,
+    &KSCATEGORY_CAPTURE,
+    &KSCATEGORY_COMMUNICATIONSTRANSFORM,
+    &KSCATEGORY_DATACOMPRESSOR,
+    &KSCATEGORY_DATADECOMPRESSOR,
+    &KSCATEGORY_DATATRANSFORM,
+    &KSCATEGORY_FILESYSTEM,
+    &KSCATEGORY_INTERFACETRANSFORM,
+    &KSCATEGORY_MEDIUMTRANSFORM,
+    &KSCATEGORY_MIXER,
+    &KSCATEGORY_RENDER,
+    &KSCATEGORY_SPLITTER,
+};
+
+static
+VOID
+Test_IoOpenDeviceInterfaceRegistryKey(
+    _In_opt_ PCWSTR SymbolicLink)
+{
+    UNICODE_STRING KeyName, SymbolicLinkName;
+    HANDLE DeviceInterfaceKey;
+    size_t n;
+
+    RtlInitUnicodeString(&SymbolicLinkName, SymbolicLink);
+    RtlInitUnicodeString(&KeyName, L"ReactOS_kmtest");
+
+    /* It's okay to call this from a user process's thread */
+    NTSTATUS Status = IoOpenDeviceInterfaceRegistryKey(&SymbolicLinkName, KEY_CREATE_SUB_KEY, &DeviceInterfaceKey);
+
+    ok(NT_SUCCESS(Status), "IoOpenDeviceInterfaceRegistryKey(): fail: 0x%x\n", Status);
+    if (!NT_SUCCESS(Status))
+    {
+        return;
+    }
+    else
+    {
+        trace("IoOpenDeviceInterfaceRegistryKey(): success: %p\n", DeviceInterfaceKey);
+    }
+
+    for (n = 0; n < RTL_NUMBER_OF(Types); ++n)
+    {
+        HANDLE DeviceInterfaceSubKey;
+        OBJECT_ATTRIBUTES ObjectAttributes;
+
+        /* Try to create the non-volatile subkey to check whether the parent key is volatile */
+        InitializeObjectAttributes(&ObjectAttributes,
+                                   &KeyName,
+                                   OBJ_CASE_INSENSITIVE | OBJ_KERNEL_HANDLE,
+                                   DeviceInterfaceKey,
+                                   NULL);
+        Status = ZwCreateKey(&DeviceInterfaceSubKey,
+                             KEY_WRITE,
+                             &ObjectAttributes,
+                             0,
+                             NULL,
+                             REG_OPTION_NON_VOLATILE,
+                             NULL);
+
+        ok(NT_SUCCESS(Status), "ZwCreateKey(): failed to create a subkey: %d 0x%x\n", n, Status);
+        if (!NT_SUCCESS(Status))
+        {
+            continue;
+        }
+        else
```

Comment:
```suggestion
            continue;
```
and you could remove the `{}` below and reduce indent level.

Reply:
Done. Same for other appropriate parts.